consensus/abci.go

```diff
@@ -244,15 +241,14 @@ func (abci *TendermintABCI) DeliverTx(txBytes []byte) tmtAbciTypes.ResponseDeliv
 
 	abci.logger.Info("Delivering TX", "hash", tx.Hash().String(), "nonce", tx.Nonce(), "cost", tx.Cost(), "gas", tx.Gas(), "height", abci.db.GetBlockStateHeader().Number.Uint64())
 
-	res := abci.db.ExecuteTx(tx)
-	if res.IsErr() {
-		abci.logger.Error("Error delivering TX to DB", "hash", tx.Hash().String(), "err", res.Log)
+	err = abci.db.ExecuteTx(tx)
```

Comment:
[comment] Database.go was returning abci responses. I refactored to improve the separation of responsabilities